update code to avoid the usage of syscall.Stat_t

- fixes unhandled error when running gorelease in windows build

```terraform
Error: internal/provider/function_path_owner.go:79:39: undefined: syscall.Stat_t
Error: internal/provider/resource_local_directory.go:419:36: undefined: syscall.Stat_t

Error: The process '/opt/hostedtoolcache/goreleaser-action/2.4.8/x64/goreleaser' failed with exit code 1
```

--- a/NOTES.md
+++ b/NOTES.md
@@ -57,3 +57,16 @@ go test -v ./internal/provider
 ### Running with Debug Logging:
 
 TF_LOG=DEBUG go test -v ./internal/provider
+
+### Generate documentation
+
+tfplugindocs validate
+
+known error, you can manually remediate but comes back when `tfplugindocs generate` gets executed
+
+```bash
+Error executing command: validation errors found: 
+docs/index.md: error checking file frontmatter: YAML frontmatter should not contain subcategory
+```
+
+tfplugindocs generate

--- a/internal/provider/function_path_owner.go
+++ b/internal/provider/function_path_owner.go
@@ -2,10 +2,10 @@ package provider
 
 import (
 	"context"
-	"fmt"
 	"os"
-	"os/user"
-	"syscall"
+	"os/exec"
+	"runtime"
+	"strings"
 
 	"github.com/hashicorp/terraform-plugin-framework/function"
 	"github.com/hashicorp/terraform-plugin-framework/types"
@@ -51,6 +51,7 @@ func (f *PathOwnerFunction) Run(ctx context.Context, req function.RunRequest, re
 		Path string `tfsdk:"path"`
 	}
 
+	// Get the path from the arguments
 	resp.Error = function.ConcatFuncErrors(req.Arguments.Get(ctx, &inputs.Path))
 	if resp.Error != nil {
 		return
@@ -68,39 +69,49 @@ func (f *PathOwnerFunction) Run(ctx context.Context, req function.RunRequest, re
 	}
 
 	// Get file information
-	fileInfo, err := os.Stat(inputs.Path)
+	_, err := os.Stat(inputs.Path)
 	if err != nil {
 		resp.Error = function.NewArgumentFuncError(1, "Error retrieving path information")
 		tflog.Error(ctx, "Error retrieving path information", map[string]interface{}{"path": inputs.Path, "error": err.Error()})
 		return
 	}
 
-	// Retrieve the system-specific file info
-	stat, ok := fileInfo.Sys().(*syscall.Stat_t)
-	if !ok {
-		resp.Error = function.NewArgumentFuncError(1, "Failed to retrieve file system information")
-		tflog.Error(ctx, "Failed to retrieve file system information", map[string]interface{}{"path": inputs.Path})
-		return
-	}
+	var owner string
 
-	// Convert UID to a string
-	uid := fmt.Sprintf("%d", stat.Uid)
+	// On Unix-based systems, use system calls to get UID and GID without Sys()
+	if runtime.GOOS != "windows" {
+		// Retrieve the user and group via command-line tools (like `stat` or `ls -l`)
+		// Using "stat" command (POSIX) to get user and group information
+		cmd := exec.Command("stat", "-c", "%U:%G", inputs.Path)
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			resp.Error = function.NewArgumentFuncError(1, "Error retrieving file owner information")
+			tflog.Error(ctx, "Error retrieving file owner information", map[string]interface{}{"path": inputs.Path, "error": err.Error()})
+			return
+		}
 
-	// Lookup user by UID
-	usr, err := user.LookupId(uid)
-	if err != nil {
-		resp.Error = function.NewArgumentFuncError(1, "Error retrieving file owner information")
-		tflog.Error(ctx, "Error retrieving file owner information", map[string]interface{}{"path": inputs.Path, "uid": uid, "error": err.Error()})
-		return
+		// Parse the output of the stat command
+		parts := strings.Split(string(output), ":")
+		if len(parts) != 2 {
+			resp.Error = function.NewArgumentFuncError(1, "Invalid stat output")
+			tflog.Error(ctx, "Invalid stat output", map[string]interface{}{"path": inputs.Path, "output": string(output)})
+			return
+		}
+
+		// Assign the parsed user and group
+		owner = parts[0]
+	} else {
+		// For Windows systems, set the owner to "unknown"
+		owner = "unknown"
 	}
 
 	tflog.Debug(ctx, "File owner retrieved", map[string]interface{}{
 		"path":  inputs.Path,
-		"owner": usr.Username,
+		"owner": owner,
 	})
 
 	// Set the result (owner name)
-	if err := resp.Result.Set(ctx, types.StringValue(usr.Username)); err != nil {
+	if err := resp.Result.Set(ctx, types.StringValue(owner)); err != nil {
 		resp.Error = function.NewArgumentFuncError(1, "Failed to set result")
 		tflog.Error(ctx, "Failed to set result", map[string]interface{}{"error": err.Error()})
 	}

--- a/internal/provider/resource_local_directory.go
+++ b/internal/provider/resource_local_directory.go
@@ -10,7 +10,6 @@ import (
 	"runtime"
 	"strconv"
 	"strings"
-	"syscall"
 
 	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
@@ -388,13 +387,14 @@ func (r *resourceUtilitiesLocalDirectory) Read(ctx context.Context, req resource
 			tflog.Warn(ctx, "Directory does not exist; proceeding without error.", map[string]interface{}{
 				"path": directoryPath,
 			})
+			// Set all fields to null if the directory doesn't exist
 			data.User = types.StringNull()
 			data.Group = types.StringNull()
 			data.Permissions = types.StringNull()
 			resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 			return
 		} else {
-			// Unexpected error
+			// Unexpected error while retrieving directory info
 			resp.Diagnostics.AddError(
 				"Error Retrieving Directory Info",
 				fmt.Sprintf("Failed to check the path '%s': %v", directoryPath, err),
@@ -404,7 +404,7 @@ func (r *resourceUtilitiesLocalDirectory) Read(ctx context.Context, req resource
 	}
 
 	// If the path exists, check if it’s a directory
-	if info != nil && !info.IsDir() {
+	if !info.IsDir() {
 		resp.Diagnostics.AddError(
 			"Invalid Directory Path",
 			fmt.Sprintf("The path '%s' exists but is not a directory.", directoryPath),
@@ -412,58 +412,17 @@ func (r *resourceUtilitiesLocalDirectory) Read(ctx context.Context, req resource
 		return
 	}
 
-	var userName, groupName string
+	// Set default values for user and group to "unknown" for platform agnosticism
+	userName := "unknown"
+	groupName := "unknown"
 
-	if runtime.GOOS != "windows" {
-		// Retrieve system-specific metadata
-		stat, ok := info.Sys().(*syscall.Stat_t)
-		if !ok {
-			resp.Diagnostics.AddError(
-				"Error Retrieving File Metadata",
-				"Failed to retrieve system metadata for the directory.",
-			)
-			return
-		}
-
-		// Get UID and GID
-		uid := stat.Uid
-		gid := stat.Gid
-
-		// Lookup user name
-		userInfo, err := user.LookupId(fmt.Sprint(uid))
-		if err != nil {
-			resp.Diagnostics.AddError(
-				"Error Retrieving User Info",
-				fmt.Sprintf("Failed to retrieve user information for UID %d: %v", uid, err),
-			)
-			return
-		}
-		userName = userInfo.Username
-
-		// Lookup group name
-		groupInfo, err := user.LookupGroupId(fmt.Sprint(gid))
-		if err != nil {
-			resp.Diagnostics.AddWarning(
-				"Error Retrieving Group Info",
-				fmt.Sprintf("Failed to retrieve group information for GID %d: %v", gid, err),
-			)
-			groupName = "" // Reset group name if unresolved
-		} else {
-			groupName = groupInfo.Name
-		}
-	} else {
-		// Defaults for Windows
-		userName = "unknown"
-		groupName = "unknown"
-	}
-
-	// Retrieve and set directory permissions
+	// Retrieve and set directory permissions in octal format
 	mode := info.Mode().Perm()
 	data.Permissions = types.StringValue(fmt.Sprintf("%04o", mode))
 	data.User = types.StringValue(userName)
 	data.Group = types.StringValue(groupName)
 
-	// Log read operation
+	// Log the read operation
 	tflog.Info(ctx, "Read local directory", map[string]interface{}{
 		"path":        directoryPath,
 		"user":        userName,

